### DISCUSSION TITLE
MailService should be catching errors? Can it state when emails were really sent?

### DISCUSSION THREAD

**Discussion by luanmm** [directus/directus#6537] [Feature Requests]
Hello, people!

First of all, thanks for all the efforts regarding Directus platform. I must congratulate all the collaborators of this really great ecosystem you're creating here!

My discussion here is about MailService. I was working with it to send e-mails in the hook "items.create.before", to mimic some trivial scenario on websites: the user submit a form and the content is sent to some mailbox. But, with Directus, there is one thing that is great: the form fields can be stored in the database (in this case it serves as "backup" or to be processed by any users that access the CMS).

But, in this context, I found out that MailService catches errors if email could not be sent, just warning in the console logger. And this is something that I was not expecting, because I can't know if the message was sent in my hook code.

In my use case, I prefer to not save the item if the email message could not be sent, to ensure that users are really receiving the form submission details in its mailboxes.

Could the try { ... } catch { ... } block be removed to allow the calling function handle it? Or maybe the "send" method could return, at least, a boolean value indicating if the email has been sent successfully?

Thanks in advance, great people! Congratulations one more time for all your efforts here! Directus is a great platform (its architecture, technologies, community, way of thinking, usability, and so on)!

Best regards!

**Comment by rijkvanzanten** [comment]
Hi @luanmm! Thanks for the kind words ❤️ 

> Could the try { ... } catch { ... } block be removed to allow the calling function handle it? Or maybe the "send" method could return, at least, a boolean value indicating if the email has been sent successfully?

Yeah, that makes sense to me! Wanna open a PR for that? I think removing the try/catch there makes most sense, as that would also allow you to check the actual error and handle it accordingly 👍🏻

**Comment by luanmm** [comment]
PR created at #6541!

Thanks for replying so quickly!

**Reply by rijkvanzanten** [reply]
Awesome, thank you!

In terms of release changelog, would you consider this an API improvement, or bug fix? 🤔

**Reply by luanmm** [reply]
Hmmm, sorry! I think my comments in PR/commit state it as a fix, but I think it is an improvement, indeed!

It is a "fix" only in my specific case/context, but I don't think it should be considered as a bug, in fact.

Thanks for considering my opinion, but I think this decision is up to the team :slightly_smiling_face:

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
